EIPS/eip-7793.md

```diff
@@ -12,37 +12,45 @@ created: 2024-10-17
 
 ## Abstract
 
-This EIP proposes to add a new precompile that returns the index of the transaction being executed within the current block.
+This EIP proposes to add a new opcode `ASSERTTXINDEX` that reverts if the index of the current transaction within the block does not match the argument provided. This allows for conditional transactions that can only be executed at a certain index within a block.
 
 ## Motivation
 
-The new precompile aims to improve support for encrypted mempools. In order to be secure, the validity of encrypted mempool transactions should be tied to the inclusion of all transactions by a proposer in the correct slot, and following the ordering rules. If these rules are not enshrined as block validity conditions then they can be enforced by a smart contract.
-
-This proposal enables smart contract solutions to check their own transaction index, so they can enforce inclusion at the correct index. These out-of-protocol solutions could be used for experimentation until a design appropriate for enshrinement is agreed upon.
+The new opcode aims to improve support for encrypted mempools. Transactions in an encrypted mempool are ordered while the transactions are encrypted, before being decrypted and included onchain at the top of the block. If the builder does not respect the order when including the decrypted transactions then they could frontrun decrypted transactions. The new opcode can be used to make this impossible; if a decrypted transaction is not included in the correct order it will revert.
 
 ## Specification
 
-If `block.timestamp >= TBD` a new precompiled contract `TXINDEX` shall be created at address `TBD`.
+If `block.timestamp >= TBD`, a new opcode `ASSERTTXINDEX` shall be created at `TBD`. It takes as an argument the specified transaction index as an offset into calldata.
+
+### Input
+
+Takes as input a 32 byte index into calldata, matching the semantics of `CALLDATALOAD`.
```

Comment:
clarify where this input comes from. it should be an immediate in the bytecode imo. instead of being read from the stack or the memory. not sure if that is possible in the EVM or not. double check